rados: fix segfaults when run without --pool

ddiss opened the pull request:
Fix and test for http://tracker.ceph.com/issues/17684 .

tchaikov commented on src/tools/rados/rados.cc:
```diff
@@ -1898,6 +1913,11 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
   }
   if (snapid != CEPH_NOSNAP) {
     string name;
```
could you move this definition down to where it is firstly used?

smithfarm commented:
`Reviewed-by: Nathan Cutler <ncutler@suse.com>`

ddiss commented:
thanks for the feedback.

changes since last version
- move name string declaration down from the start of the code block
- add Nathan's reviewed-by tags

smithfarm commented:
@ddiss Actually, the "Reviewed-by" tag is supposed to go into the merge commit, signifying that the PR as a whole was reviewed. So it's the reviewer who's supposed to add it when the PR is merged.

ddiss commented:
@smithfarm okay, pushed again without your tags

tchaikov commented:
lgtm

ddiss commented:
The Jenkins TestErasureCodePluginJerasure.cc segfault doesn't appear to be in any way related to this change.

smithfarm commented:
Jenkins test this please

smithfarm commented:
In all likelihood, the jerasure plugin test failure is http://tracker.ceph.com/issues/17561

tchaikov commented:
- http://pulpito.ceph.com/kchai-2016-11-10_07:59:52-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2016-11-10_02:18:01-rados-wip-kefu-testing---basic-mira/
